### ISSUE TITLE
Running app in production.

### ISSUE THREAD

**Issue by FahadMullaji** [angular/universal-starter#202]
Ok. So I am done with all the hard work of converting my Angular 1 app to Angular 2 app and then to Universal app. But now I am struggling with deploying it AWS server.

I get the part where I have to run `npm run build:prod` which generates minified and uglified version of whole application in '$/dist/server/index.js'. which is awesome.

But when I go to that location ('$/dist/server/') and run `node index.js` or `nodemon index.js` and access localhost:8080 then I get 

![universal issue](https://cloud.githubusercontent.com/assets/1390630/19605647/978ec630-9781-11e6-9b73-18bfa41fcb02.png)

But when I go to root location of app and run `node dist\server\index.js` or `nodemon dist\server\index.js` and access localhost:8080 then it works fine. What's up with that?
I will have to push all the code to AWS even after building app to index.js?

Let's say that I can live with that. 
But again when I try to run application in the background using either `forever` or `pm2` then I get the same error which I have mentioned above.
does not matter if I do 
`forever start index.js` or `forever start dist\server\index.js`
`pm2 start index.js` or `pm2 start dist\server\index.js`

So even If I have to push all the code to AWS, I can't run it in background.

Maybe I am doing something wrong. Or all my concepts are not clear. 

Thanks
Fahad Mullaji

**Comment by kuldeepkeshwar** [comment]
Webpack set the _dirname to the project root if webpack is configured with `_dirname:true`
Overriding the default value of _dirname provided by node

**Comment by aiboy** [comment]
same here.. can't run universal-starts in production mode (same error). Can anyone at least tell where to look?

**Comment by FahadMullaji** [comment]
@aiboy So I pushed all my code to production server and ran `pm2 start dist\server\index.js`, which runs the application in the background.

I think I had to update pm2 package to make it work. I would suggest the same.
Plus I am running the application on port 8080 and I have reverse proxied my apache server so when you access the link apache passes the request to express server.

I still don't why it was not working before it is working now but I still do not like the part where I have to move all my code when I build application to index.js

I know universal has process to push application to Heroku but they should have efficient process to push code to AWS too. So I am keeping this issue open.
